pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -465,6 +467,9 @@ def update_logger_connector(self, fx_name: str = None) -> None:
         logger_connector.callback_metrics.update(callback_metrics)
         logger_connector.callback_metrics.pop("epoch", None)
 
+        batch_pbar_metrics.pop("debug_epoch", None)
```

Comment:
Can you explain why this is needed?

Reply:
It will break the tests if not being popped :) 
debug_epoch is being injected and I need to remove it from callback_metrics.

Reply:
Where is it being injected?

To clear the testing state inside the tested code seems quite ugly.